lib/concurrent-ruby/concurrent/atomic/java_thread_local_var.rb

```diff
@@ -1,37 +0,0 @@
-require 'concurrent/atomic/abstract_thread_local_var'
-
-if Concurrent.on_jruby?
-
-  module Concurrent
-
-    # @!visibility private
-    # @!macro internal_implementation_note
-    class JavaThreadLocalVar < AbstractThreadLocalVar
```

Comment:
There were various discussions in the past whether it's better to keep this or not.
I think given that's actually per-Fiber on JRuby it's just so confusing, so agreed let's simplify and use the Ruby ones for all.